### DISCUSSION TITLE
Why should I use this over zdharma-continuum/zinit?

### DISCUSSION THREAD

**Discussion by pd93** [z-shell/zi#138] [Q&A] [answered]
When the zdharma org was deleted, I jumped over to [zdharma-continuum/zinit](github.com/zdharma-continuum/zinit) along with many other zinit users and haven't thought about again since. However, a friend of mine recently migrated from the original repo and was directed here by the [README in zdharma/zinit](https://github.com/zdharma/zinit) which links directly to zi. This led me down a bit of a rabbit hole and I have a few questions.

- Why was this repo created? - more precisely, what is it that [zdharma-continuum/zinit](github.com/zdharma-continuum/zinit) doesn't do this made this project necessary and why should I use this instead?
- Why was the git history scrubbed?
  - I'm aware that the original project did not require any attribution and was open source, but this seems like a bit of a strange decision. Even if credit is not required, it can seem suspicious to users if an author intentionally made an effort to remove contribution history from a project.
- Who currently owns the [zdharma](github.com/zdharma) org?
  - The README links to this project and so is obviously associated with your project in some way. However, the original zdharma org was deleted and the new one was created around the same time as this project. It is also owned by a new user with no other projects or contributions making the redirect difficult to trust. Without speculating too much, their username is also somewhat similar to the previous org owner who deleted their account.

To be clear, I'm not making any accusations or attributing any blame. However, I would appreciate some clarity around these points and perhaps some better communication with the community regarding some of the decisions made.

**Comment by ss-o** [comment, marked as answer]
> Why was this repo created? - more precisely, what is it that [zdharma-continuum/zinit](https://github.com/orgs/z-shell/discussions/github.com/zdharma-continuum/zinit) doesn't do this made this project necessary and why should I use this instead?

I am not going to make comments about other projects - all the best to all :)

This indirectly answers the questions:
> From: https://github.com/z-shell/zi#acknowledgements
The [Z-Shell](https://github.com/z-shell) was created to recover the zdharma organization project which was deleted by the owner. We don't want to depend on an unreliable source. [ZI](https://github.com/z-shell/zi), formerly known as zplugin, zinit, is an open-source community project released under the [MIT License](https://github.com/z-shell/zi/blob/main/LICENSE). 

 - [x] accomplished

Could you provide more details on "a bit of a rabbit hole"? zdharma-continuum, seems to be a popular project - why switch?

> Why was the git history scrubbed? I'm aware that the original project did not require any attribution and was open source, but this seems like a bit of a strange decision. Even if credit is not required, it can seem suspicious to users if an author intentionally made an effort to remove contribution history from a project.

History can be recovered at any point in time and any file also be appended or prepended. I could not find any of your commits. If I missed it and it matters to you, just let me know. As far as I am concerned time is what matters most. I have every single commit of all zdharma and psprint history + including all deleted, and not only on GitHub.



> Who currently owns the [zdharma](https://github.com/orgs/z-shell/discussions/github.com/zdharma) org? The README links to this project and so is obviously associated with your project in some way. However, the original zdharma org was deleted and the new one was created around the same time as this project. It is also owned by a new user with no other projects or contributions making the redirect difficult to trust. Without speculating too much, their username is also somewhat similar to the previous org owner who deleted their account.

After zdharma's disappearance and part of recovery (as stated previously) also due to many visitors from the east using cached links the first step taken was to investigate the most vulnerable spots and create strategies to secure them. I shared it with colleagues and we secured and arranged automation to cover future changes and surprises. Any domain name or service with 'zdharma' 'zinit' around the globe is owned by my team.

Also, much information was recovered about the previous owner and the project, which is private and can not be shared.

Anyone is welcome to the project - open-source is where people create and share mostly because of their passion. :octocat: Choose the project you like and make it better for me there is no difference in the choise :1st_place_medal:.

https://github.com/z-shell/.github/tree/main/governance
https://github.com/z-shell/community/issues/new?assignees=&labels=%F0%9F%91%A5+member&template=membership.yml&title=team%3A+

Welcome for any additional questions. :upside_down_face: 

P.S: Anything missing/changed can be recovered if there is a legit reason for it. Communication can help to solve it :+:

Sincerely,

**Reply by pskarma** [reply]
Forgot to mention, that project was created before zdharma was even deleted and way before all continumes. Because we didn't like the activity of the project. This was probably the main cause of history deletion and etc., - to deduplicate it.

**Reply by pd93** [reply]
First of all, thanks for taking the time to write a detailed reply.

> I am not going to make comments about other projects - all the best to all :)

That's fair enough, though I was more asking about what your project has to offer than the shortcomings of continuum. If I can't split hairs between two similar projects, I tend to use the one that is more popular, and at the moment, that is continuum. I was simply wondering if there was a good reason why z was a better choice.

> Could you provide more details on "a bit of a rabbit hole"?

The rabbit hole I am referring to are the [posts on GitHub and/or Reddit](https://www.reddit.com/r/zsh/comments/sqsni7/zshellzi_vs_zdharmacontinuumzinit/) which speculate about the reasons behind the deletion and discuss whether or not either fork can be trusted. I also read through the [reddit post](https://www.reddit.com/r/zsh/comments/qinb6j/httpsgithubcomzdharma_has_suddenly_disappeared_i/) from the day the original repo was deleted and found no mention of z-shell and had discussions with several colleagues who all had the same questions as me. It wasn't a particularly deep hole, but that's why I am still digging. I believe its better to ask you directly, than to speculate elsewhere.

> zdharma-continuum, seems to be a popular project - why switch?

This is precisely why I am asking these questions. As it stands, I have no reason to switch.

> History can be recovered at any point in time and any file also be appended or prepended. I could not find any of your commits. If I missed it and it matters to you, just let me know. As far as I am concerned time is what matters most. I have every single commit of all zdharma and psprint history + including all deleted, and not only on GitHub.

I respectfully disagree. Indeed, history can be recovered, and in fact, continuum has a complete history of the project. I have not made any contributions to ZSH or zinit and wouldn't even know where to start. It's more that if you like a project - enough to make a fork, than I think you should respect the authors that have come before you and the contributions that they (and the community) have made. Historians do not delete things from history books because they don't like it. They are a part of the history of the project and should be preserved. Moreover, it makes it look like you have taken claim over all these commits - even if that was not the intention. I believe this is why the "fork" functionality exists and is the accepted way to respectfully continue another person's work.

> After zdharma's disappearance and part of recovery (as stated previously) also due to many visitors from the east using cached links the first step taken was to investigate the most vulnerable spots and create strategies to secure them. I shared it with colleagues and we secured and arranged automation to cover future changes and surprises. Any domain name or service with 'zdharma' 'zinit' around the globe is owned by my team.

I'm still not sure I understand how you can recover all the accounts of the original user without help from the original user themselves. Perhaps this is just my ignorance, but it feels like if it were that easy, then many others would've attempted the same thing. 

Ultimately, I guess this boils down to a couple of things for me:

1. I personally find it hard to trust something that has taken over an org that I already didn't trust without much explanation as to how they acquired it or what their intentions for it are. "[Zinit Legacy. Use ZI instead](https://github.com/zdharma/zinit)" is not enough for me.
2. The removal of the Git history compounds the above and lowers my trust.

This is just my gut feeling and opinion. I'm sure that your intentions are genuine and wish you all the best with this project. But I can't see how I am able to use it going forwards without these issues being addressed.

> Anyone is welcome to the project - open-source is where people create and share mostly because of their passion. :octocat: Choose the project you like and make it better for me there is no difference in the choise

This is 100% something we can agree on. As I said before, thank you for taking the time to reply. I hope this has been constructive for you rather than critical. It's never my intention to criticise or annoy someone who is taking the time and effort to contribute useful tools to the community.

All the best,

**Reply by ss-o** [reply]
With a second message I understood you as well :handshake: 

Sometimes, I get overwhelmed, as my background is in cybersecurity. I am barely sociable, as social media is good tool and nothing more. I know what's going on around me - after starting @z-shell the number of trolls increased. I assume, most would get a bit pissed off when someone requires attention with no purpose or reason.

My personal advice is don't choose the most popular projects all the time :octocat: Take some time to understand not how open source works in general, but the idea of it and why there are no "best" and rarely "better" projects. Then choose a few projects based on the idea and try to participate. Most likely you will thank me later.

Finally, as I mentioned before everyone here. :heart: 

Thank you for taking the time to write a message, hopefully, it helped.

Take care mate,

**Reply by Freed-Wu** [reply]
> Historians do not delete things from history books because they don't like it. They are a part of the history of the project and should be preserved.

:+1: History can help us find a bug or a change's origin, which can let us understand why the code is written in this method at the beginning. It is very important.

> History can be recovered at any point in time and any file also be appended or prepended. I could not find any of your commits

Can history be recovered? I have commit f399defbaebd539aa9fb2a9b92d1824475d8caf0 :smile:

> I am barely sociable, as social media is good tool and nothing more

I advise to socially communicate with @zdharma-continuum, to see anything both of them can cooperate. It can help them not fix a bug twice, and get some same features.

**Comment by anwalk** [comment]
:us: Make ZI Great Again! :us:

**Comment by pskarma** [comment]
Also, this user just copying code from ZI repos:

https://github.com/zdharma-continuum/zinit/pull/173/files
https://github.com/zdharma-continuum/zinit/commit/42e83d7f99254c16e408f52848b914f7aa264372
https://github.com/zdharma-continuum/zinit/commit/047320a9234be4de8299ff4796e28e2363e77984

Other commits similar or just CI/Docs/Etc.

Also compared overall internal functions and compatibility with repos (annex, packages, etc.)

1 improvement made - jq for packages. (good improvement), but compatibility with repos low, some functions/APIs do not match (simply not working). ZI has similar issues almost fixed - 90% of repos under z-shell are up to date and working. Many improvements come after fixes are done.

Not much to compare.

**Reply by ss-o** [reply]
Don't be so sensitive :speak_no_evil:

**Comment by aidan-gibson** [comment]
For future lurkers trying to make the same decision:
* zdharma-continuum mostly appears to be a preservation project so zinit would keep working for those using the software before zdharma left. Innovation / new features are limited AFAIK.
* z-shell/zi isn't shackled by the past and can freely continue to innovate. 

New users just discovering zi/zinit should use zi.

The points pd93 makes here are valid; if unsure, I also default to repos with more stars, as they're generally better maintained. However, that is not the case here. 

Looking at the overall contributions can be misleading, as zdharma-continuum includes commits from the original project and z-shell/zi does not. If you filter by when z-shell/zi started, you can see much more work is put into z-shell/zi than continuum:
![Pasted image 20230103144601](https://user-images.githubusercontent.com/20917687/210456274-fccda08e-beb6-47e0-817b-9648a41b0b16.png)
![Pasted image 20230103144613](https://user-images.githubusercontent.com/20917687/210456293-094308ec-c247-4fb8-9f1a-dae26d35cfe6.png)

In my personal experience, when I posted an issue on z-shell/zi, @ss-o responded _immediately_, and worked his ass off to push out commit after commit to fix the issue. I was floored by his work ethic & dedication to this project, he dropped everything and worked something like 6 hours straight until my issue was resolved. 

This situation is a case study in why stars aren't everything in judging github projects. Based on the value provided, z-shell/zi deserves much more recognition than it's currently getting.

**Reply by Lockszmith-GH** [reply]
In July 2024, Is this still the case?
Asking because I see acitivty reduced here, while the other project seems to be active
